src/index.ts

```diff
@@ -86,21 +89,22 @@ ${precompiledTemplate}
 })();
 `);
 
-        for (let foundTemplateName of visitor.foundTemplateNames) {
-            // require takes module name separated with forward slashes
-            parts.push(`require('${slash(foundTemplateName)}');`);
-        }
+            for (let foundTemplateName of visitor.foundTemplateNames) {
+                // require takes module name separated with forward slashes
+                parts.push(`require('${slash(foundTemplateName)}');`);
+            }
 
-        parts.push(`env.registerTemplatesModule(templatesModule, '${key}');`);
+            parts.push(`env.registerTemplatesModule(templatesModule, '${key}');`);
 
-        parts.push(`
+            parts.push(`
 let template = env.loadTemplate('${key}');
 
 module.exports = (context = {}) => {
-    return template.render(context);
+    return template.then((template) => template.render(context));
```

Comment:
The module that the webpack loader returns should not be returning a Promise. This will break a whole bunch of integrations like Storybook.

And there's no need to have the module return an async function. `template.render()` is a sync function. At the end of `visitor.visit(module)` Promise chain, you can create a module with a sync function that renders the template.

See visitor.visit(module)

Reply:
The only weird bit of the code snippet I linked to is line 136 `env.emit('template', name);` because I had to do the same things that `env.loadTemplate()` would normally do.

Reply:
`TwingTemplate::render` is an async function. Do you have an idea on how to transform it to a sync function?

About Storybook, do you mean that they assume that loaders returns sync functions? If so, it's a bold asumption, and a debatable one. They should ensure at their level that they wrap everything with a promise, using `Promise.resolve`. Is there an issue about that point in their repository?

Reply:
You know what? It's after midnight here, so I think I'm mis-remembering something. It was the PR at https://github.com/NightlyCommit/twing-loader/pull/24 that was breaking Storybook. I should test the change you made with Storybook.

Thanks for making these changes! I'll try to test this in the next few days.

Reply:
Let me know how it goes. I'm a little blind here since I don't use either webpack or storybook. So your feedbacks are all I have. :D